.github/workflows/rust.yml

```
@@ -10,70 +9,31 @@ env:
   CARGO_TERM_COLOR: always
 
 jobs:
-  rust-test:
+  rust-build-test:
     runs-on: ubuntu-20.04
     steps:
       - name: Checkout branch
         uses: actions/checkout@v2
-      - name: Set up Rust caches
-        uses: actions/cache@v2
-        id: rust-cache
         with:
-          path: |
-            ~/.cargo/bin/
-            ~/.cargo/registry/index/
-            ~/.cargo/registry/cache/
-            ~/.cargo/git/db/
-            target/
-          key: ${{ runner.os }}-cargo-${{ hashFiles('Cargo.lock') }}
-      - name: Run steward acceptance tests
-        run: cargo test --all --verbose
-
-  rust-build:
-    runs-on: ubuntu-20.04
-    steps:
-      - name: Checkout branch
-        uses: actions/checkout@v2
+          profile: minimal
+          toolchain: stable
+          override: true
       - name: Set up Rust caches
-        uses: actions/cache@v2
+        uses: actions/cache@v3
         id: rust-cache
         with:
           path: |
-            ~/.cargo/bin/
-            ~/.cargo/registry/index/
-            ~/.cargo/registry/cache/
-            ~/.cargo/git/db/
-            target/
-          key: ${{ runner.os }}-cargo-${{ hashFiles('Cargo.lock') }}
-      - name: Run Cargo build
-        run: cargo build
-
-  fmt:
-    name: Rustfmt
-    runs-on: ubuntu-20.04
-    steps:
-      - name: Checkout branch
-        uses: actions/checkout@v2
-        with:
-          profile: minimal
-          toolchain: stable
-          override: true
+            ~/.cargo/registry
+            ~/.cargo/git
+            target
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
       - name: Add rustfmt
```

I think these were separated out because it makes it clearer in Github Actions which aspect failed, rather than having to look at the logs. We might want to keep it that way.

---

It also allows them to run in parallel if they aren't depending on each other